### PR TITLE
Fix system property resolution at configuration time

### PR DESCRIPTION
**Pull request by breskeby**
This fixes system property resolution when running the build with `--configuration-cache`

One step closer to make use of configuration cache

### CODE REVIEW

**Comment by breskeby** [comment]
This fixes system property resolution when running the build with --configuration-cache

One step closer to make use of configuration cache

File: build-conventions/src/main/java/org/elasticsearch/gradle/internal/conventions/VersionPropertiesLoader.java [hunk 1]
```diff
@@ -43,19 +43,17 @@ protected static void loadBuildSrcVersion(Properties loadedProps, ProviderFactor
             );
         }
         String qualifier = providers.systemProperty("build.version_qualifier")
-                .orElse("")
                 .forUseAtConfigurationTime()
-                .get();
+                .getOrElse("");
```
Comment (pugnascotia) [inline build-conventions/src/main/java/org/elasticsearch/gradle/internal/conventions/VersionPropertiesLoader.java:47]: For my own education, why is this way better?

**Review by pugnascotia — APPROVED** [review]

File: build-conventions/src/main/java/org/elasticsearch/gradle/internal/conventions/VersionPropertiesLoader.java (hunk 1 above)
Comment (breskeby) [reply build-conventions/src/main/java/org/elasticsearch/gradle/internal/conventions/VersionPropertiesLoader.java:47]: the forUseAtConfigurationTime() unfortunately isn't applied to the Provider returned I the `orElse` block, which makes the build fail when running with `configuration-cache` enabled. That behaviour is quite unexpected, at least for my understanding. I'm discussing the shortages of this api with the Gradle team

Comment (breskeby) [reply build-conventions/src/main/java/org/elasticsearch/gradle/internal/conventions/VersionPropertiesLoader.java:47]: having said that the gradle team mentioned that `forUseAtConfigurationTime` will go away in the future

Comment (mark-vieira) [reply build-conventions/src/main/java/org/elasticsearch/gradle/internal/conventions/VersionPropertiesLoader.java:47]: > having said that the gradle team mentioned that forUseAtConfigurationTime will go away in the future

Is there going to be a replacement? Does that mean that you can't depend on things like system properties or environment variables at configuration time?

**Pull request merged**